File: elixir/lib/pagerduty_client.ex
defmodule PagerdutyClient do
  def main(_args) do
    IO.puts("This is where the execution code can be added")
  end

  def http_get(url) do
    headers = %{"User-Agent" => "Mozilla/5.0"}

    IO.puts("\nSending 'GET' request to URL : #{url}")
    response = HTTPoison.get!(url, headers)
    IO.puts("Response Code : #{response.status_code}")
    response.body
  end

  def json_to_map(json_string) do
    JSON.decode!(json_string)
  end

  def get_input_line() do
    IO.read(:line) |> String.trim()
  end
end

File: python/pagerduty_client.py
import requests

class PagerdutyClient:
    @staticmethod
    def http_get(url):
        headers = {
            "User-Agent": "Mozilla/5.0"
        }
        print(f"\nSending 'GET' request to URL : %s" %(url))
        response = requests.get(url, headers=headers)

        print("Response Code : %s" %(response.status_code))
        print(response)
        return response.text

    @staticmethod
    def get_input_line():
        return input()

if __name__ == "__main__":
    print("This is where the execution code can be added")
